### PR TITLE
Adds and Replaces To_Chat() With Balloon_Alerts() in Drone, Hivelord, and Queen Abilities

### PR DESCRIPTION
**Pull request by KornFlaks**
## About The Pull Request

Some more balloon alerts. This time for drones, and their fatter cousins.

With all these balloon alerts, we'll need to see if any accidentally conflict and overlap.

## Why It's Good For The Game

Shamelessly copy pasting what RipGrayson said in their (now closed) PR:

> We're moving away from the philosophy of needing to stare at chat in order to see what's going on, runechat did it for speech, balloon_alerts do it for actions. It's just so much easier to see what's going on by looking at your character than it is to dig around in a bunch of text.

> On the aesthetic side it actually looks really cool to see overhead notifications like "Franklin Ramis fixes the internal wiring of platinum miner" or "Queen (236) starts emitting pheremones", it makes us look very smooth.

> So by implementing this not only do we look fancy, but it also enhances team cooperation because xenos/marines can see what their teammates are doing while being focused on combat. (Not that they couldn't before, they would just have to scroll through chat, which made it impractical).

## Changelog
:cl:
qol: Drones, Hivelords, and Queens now have balloon alerts for their abilities.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm [hunk 1]
```diff
@@ -111,35 +114,38 @@
...
 	if(!do_after(owner, HIVELORD_TUNNEL_DIG_TIME, TRUE, T, BUSY_ICON_BUILD))
+		X.balloon_alert(X, "Digging aborted")
 		to_chat(owner, span_warning("Our tunnel caves in as we stop digging it."))
 		return fail_activate()
 
 	if(!can_use_action(TRUE))
 		return fail_activate()
 
-	var/mob/living/carbon/xenomorph/hivelord/X = owner
+	T.balloon_alert(X, "Dug tunnel")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm:148]:
```suggestion
	T.balloon_alert(X, "Tunnel dug")
```

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 2]
```diff
@@ -92,6 +92,7 @@
 	keybind_signal = COMSIG_XENOABILITY_SCREECH
 
 /datum/action/xeno_action/activable/screech/on_cooldown_finish()
+	owner.balloon_alert(owner, "Screech ready")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:95]: Ditto from other prs if this should be with the cooldowns

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 3]
```diff
@@ -344,7 +345,8 @@
 
 	if(xeno_ruler.xeno_caste.queen_leader_limit <= length(xeno_ruler.hive.xeno_leader_list))
 		if(feedback)
-			to_chat(xeno_ruler, span_xenowarning("We currently have [length(xeno_ruler.hive.xeno_leader_list)] promoted leaders. We may not maintain additional leaders until our power grows."))
+			xeno_ruler.balloon_alert(xeno_ruler, "No more leadership slots")
+			to_chat(xeno_ruler, span_xenowarning("We currently have the maximum of [length(xeno_ruler.hive.xeno_leader_list)] promoted leaders. We may not maintain additional leaders until our power grows."))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:349]: remove to chat here?

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 4]
```diff
@@ -353,7 +355,9 @@
 /datum/action/xeno_action/set_xeno_lead/proc/unset_xeno_leader(mob/living/carbon/xenomorph/selected_xeno, feedback = TRUE)
 	var/mob/living/carbon/xenomorph/xeno_ruler = owner
 	if(feedback)
+		xeno_ruler.balloon_alert(xeno_ruler, "Xeno demoted")
 		to_chat(xeno_ruler, span_xenonotice("We've demoted [selected_xeno] from Lead."))
+		selected_xeno.balloon_alert(selected_xeno, "Leadership removed")
 		to_chat(selected_xeno, span_xenoannounce("[xeno_ruler] has demoted us from Hive Leader. Our leadership rights and abilities have waned."))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:361]: remove to chat?

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 5]
```diff
@@ -365,10 +369,12 @@
...
 	if(!(selected_xeno.xeno_caste.caste_flags & CASTE_CAN_BE_LEADER))
 		if(feedback)
-			to_chat(xeno_ruler, span_xenowarning("This caste is unfit to lead."))
+			xeno_ruler.balloon_alert(xeno_ruler, "Xeno cannot lead")
 		return
 	if(feedback)
+		xeno_ruler.balloon_alert(xeno_ruler, "Xeno promoted")
 		to_chat(xeno_ruler, span_xenonotice("We've selected [selected_xeno] as a Hive Leader."))
+		selected_xeno.balloon_alert(selected_xeno, "Promoted to leader")
 		to_chat(selected_xeno, span_xenoannounce("[xeno_ruler] has selected us as a Hive Leader. The other Xenomorphs must listen to us. We will also act as a beacon for the Queen's pheromones."))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:378]: remove at minimum the first to chat, you could consider the second one as important to explain what it does to new players

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 6]
```diff
@@ -423,8 +429,7 @@
 	add_cooldown()
 	receiver.gain_plasma(300)
 	succeed_activate()
-	to_chat(owner, span_xenonotice("We transfer some plasma to [target]."))
-	to_chat(receiver, span_xenonotice("We feel our plasma reserves increase. Bless the Queen!"))
+	receiver.balloon_alert_to_viewers("Plasma given from Queen")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:432]: Debatable if it should be visible to all but ill leave it up to you

Comment (KornFlaks) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:432]: A sudden jump in plasma and also reminds everyone nearby about the (frankly overpowered) ability of the queen to grant large amounts of plasma.

This decision to make this message visible for everyone is coming more from my preference as a queen main. Not seeing transfer plasma used by a queen is very concerning and is indicative of an inexperienced player but detecting its usage is difficult at best.

The message now being visible will allow the more experienced players to remind the queen to occasionally use their abilities to help those around them.

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm (hunk 4 above)
Comment (KornFlaks) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:361]: Retaining the to_chats here is from personal experience.

The xeno leadership selection menu is extremely poor (on my to-do list to improve). I often accidentally select already promoted xenos for leadership, thus demoting them.

Removing the name from the demotion message would mean spending just a little more time trying to figure out who I accidentally demoted and then re-promote them before finding another xeno to promote.

Hrm, I'll get onto fixing that immediately with at least a bandaid solution. Pie in the sky is of course a TGUI hive status.

Comment (KornFlaks) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm]: Resolved in new PR.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**